src/main/java/org/example/service/impl/TraineeServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+    package org.example.service.impl;
+
+    import org.example.dao.TraineeDao;
+    import org.example.models.Trainee;
+    import org.example.service.TraineeService;
+    import org.example.util.LogUtil;
+    import org.example.util.Generator;
+    import org.example.util.UserUtil;
+    import org.springframework.beans.factory.annotation.Autowired;
+    import org.springframework.stereotype.Service;
+
+    import java.util.List;
+
+    @Service
+    public class TraineeServiceImpl implements TraineeService {
+        private final TraineeDao traineeDao;
+        private UserUtil userUtil;
+        @Autowired
+        public TraineeServiceImpl(TraineeDao traineeDao) {
+            this.traineeDao = traineeDao;
+        }
+
+        @Autowired
+        public void setUserUtil(UserUtil userUtil){
+            this.userUtil = userUtil;
+        }
+        @Override
+        public void createTrainee(Trainee trainee) {
+            LogUtil.info(TraineeServiceImpl.class, "Creating new trainee...");
+
+            List<String> existingUsernames = userUtil.getAllExistingUsernames();
+
+            LogUtil.info(TraineeService.class, "Generating new username and password...");
+            String username = Generator.generateUsername(trainee.getFirstName(), trainee.getLastName(), existingUsernames);
+            String password = Generator.generatePassword();
+            Long userId = Generator.generateId(traineeDao.listAll());
+
+            trainee.setUsername(username);
+            trainee.setPassword(password);
+            trainee.setUserId(userId);
+
+            traineeDao.create(trainee);
+        }
+        @Override
+        public void updateTrainee(Trainee trainee) {
+            LogUtil.info(TraineeServiceImpl.class, "Updating current trainee...");
+            traineeDao.update(trainee);
+        }
+        @Override
+        public void deleteTrainee(Long userId) {
+            LogUtil.info(TraineeServiceImpl.class, "Deleting trainee...");
```

Comment:
Does logger work? I launched locally and nothing printed

Reply:
I think you need to add dependency in pom for logger.
Then in every class use smg like private static final Logger LOGGER = LoggerFactory.getLogger(TraineeService.class);